pkg/logging/filelogging_test.go

```diff
@@ -0,0 +1,144 @@
+	"github.com/openshift/installer/pkg/asset/machines"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+)
+
+func convertToFormat(buf bytes.Buffer) string {
+	var dat map[string]interface{}
+
+	if err := json.Unmarshal(buf.Bytes(), &dat); err != nil {
+		return ""
+	}
+	return dat["msg"].(string)
+}
+
+func getAsset(filename string) *asset.File {
+	return &asset.File{
+		Filename: filename,
+	}
+}
+
+func TestLogFilesChanged(t *testing.T) {
+	cases := []struct {
+		name                  string
+		assets                []asset.WritableAsset
+		cmdName               string
+		directory             string
+		expectedGenerationLog string
+	}{
+		{
+			name:                  "test empty assets list",
+			assets:                []asset.WritableAsset{},
+			cmdName:               "test",
+			directory:             "test",
+			expectedGenerationLog: "",
+		},
+		{
+			name: "test asset with one file",
+			assets: []asset.WritableAsset{
+				&installconfig.InstallConfig{
+					File: &asset.File{
+						Filename: "a.yaml",
+					},
+				},
+			},
+			cmdName:               "test install config",
+			directory:             "test",
+			expectedGenerationLog: "test install config created in the test directory",
+		},
+		{
+			name: "test asset with two files, same directory",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("b.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in the test directory",
+		},
+		{
+			name: "test asset with two files, two directories",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in the test and test/machines directories",
+		},
+		{
+			name: "test asset with two files, two directories, but three entries (same file twice",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+						getAsset("machines/b.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in the test and test/machines directories",
+		},
+		{
+			name: "test asset with three files, two directories",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+						getAsset("machines/c.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test/",
+			expectedGenerationLog: "machines created in the test and test/machines directories",
```

Comment:
my preference would be just point to `test`, no reason we need to say assets in test and test/machines ? How do you think this will be useful summary?

Reply:
but i'm fine with this if we think it is more useful to the users. no need to change.